guppylang/std/_internal/compiler/array.py

```diff
@@ -180,44 +186,158 @@ def compile(self, args: list[Wire]) -> list[Wire]:
             return [self.build_classical_array(args)]
 
 
+ARRAY_GETITEM_CLASSICAL: Final[GlobalConstId] = GlobalConstId.fresh()
```

Comment:
On second thought, maybe we should store the user-facing name also inside `GlobalConstId`?